clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/model/GoogleSubnet.groovy

```diff
@@ -30,5 +30,5 @@ class GoogleSubnet implements Subnet {
   String account
   String region
   String selfLink
-  String purpose = "n/a"
+  String purpose
```

Comment:
A network for internal http lb must have a subnet whose 'purpose' is 'INTERNAL_HTTPS_LOAD_BALANCER'.

Using this field at here
https://github.com/spinnaker/deck/pull/8397/files#diff-7a244618cbb95aadb8a1890d137bab38R171

Reply:
This seems like a relatively arbitrary place to store this information. Why not a `type` field or something? Also, can you be sure that no one else will be affected by the fact that purpose is now `null` for things that aren't internal HTTP load balancers?

But also I'm confused about where this gets set? I would have expected to see the caching agent set this field, but it's not there, so where does it come from?

Reply:
`purpose` is a field of `com.google.api.services.compute.model.Subnetwork`
https://googleapis.dev/java/google-api-services-compute/latest/index.html

It's saved into cache at here.
https://github.com/spinnaker/clouddriver/blob/53d68f05ffdaebdf9cc8f6437033ee4672079138/clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleSubnetCachingAgent.groovy#L89

And passed into GoogleSubnet constructor at here.
https://github.com/spinnaker/clouddriver/blob/50521ced38c4851056032b32f338f75f8a7dbc00/clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleSubnetProvider.groovy#L81

I think this usage is not wrong..

---

I grepped `clouddriver-core`, `clouddriver-google` and `deck`, and it seems no code depends the value `n/a` of this field. 
But I agree that setting `n/a` as default value is more safe way. I'll fix it.

Reply:
fixed

Reply:
Ah, I see, I didn't realize this was coming directly from GCE. Thanks for the pointer and the fix.